src/Evolution/Systems/CurvedScalarWave/Worldtube/SingletonChare.hpp

```diff
@@ -36,19 +44,35 @@ struct WorldtubeSingleton {
   using metavariables = Metavariables;
   using evolved_vars = ::Tags::Variables<
       tmpl::list<CurvedScalarWave::Tags::Psi, CurvedScalarWave::Tags::Pi>>;
+  // not currently supported
+  static constexpr bool local_time_stepping = false;
+  using initialization_actions = tmpl::list<
+      ::Initialization::Actions::InitializeItems<
+          ::Initialization::TimeStepping<Metavariables, local_time_stepping>,
+          Initialization::InitializeElementFacesGridCoordinates<Dim>>,
+      Parallel::Actions::TerminatePhase>;
 
-  using initialization_actions =
-      tmpl::list<Initialization::InitializeElementFacesGridCoordinates<Dim>,
-                 Parallel::Actions::TerminatePhase>;
-
-  using step_actions = tmpl::list<Actions::ReceiveElementData>;
-
+  struct worldtube_system {
+    static constexpr size_t volume_dim = Dim;
+    static constexpr bool has_primitive_and_conservative_vars = false;
+    using variables_tag = ::Tags::Variables<tmpl::list<Tags::Psi0>>;
+  };
+  using step_actions =
+      tmpl::list<Actions::ReceiveElementData,
+                 Actions::SendToElements<Metavariables>,
+                 ::Actions::RecordTimeStepperData<
+                     typename worldtube_system::variables_tag>,
+                 ::Actions::UpdateU<typename worldtube_system::variables_tag>>;
```

Comment:
I'm not sure if this is because this is a work in progress or something, but it looks like after updating `Psi0` with the time stepper you immediately overwrite it in `ReceiveElementData`.  What is going on here?

Reply:
`ReceiveElementData` now overwrites the `PsiMonopole`, not `Psi0`. But yes, at the moment the time stepper is a bit redundant because at expansion orders 0 and 1 it is just needed to change the slab size which I am adding in a follow up PR. The time derivative is never changed from 0, so nothing happens.

Once I add expansion order 2, the time stepper is needed to solve an ODE in the Singleton.

Reply:
Ah, missed that that got changed in the rename.

Still a bit confused, though.  Why do you need to evolve an ODE to adjust the slab size?  Why do you care about the slab size if you're not evolving anything?

Reply:
At expansion order n = 0 or 1, I am not really evolving anything in the worldtube singleton. The elements abutting it send data to it and the singleton  sends it right back - the time stepper does not evolve anything. However, the `TimeStepId` is used as the key for the inboxes, so the singleton knows that it got the data from each neighboring element at a certain time step. When the global time step is changed, the worldtube needs to know about this and adjust its time step accordingly so the `TimeStepId` it receives from the elements is the same it is currently on. I suppose at these expansion orders, the worldtube singleton could probably get by without the time stepper but it is good to check the TimeStepId anyway.

Once I add expansion order 2, the time stepper in the worldtube will actually evolve an ODE. This ODE needs to be evolved at the same global time step as the elements so that the boundary information is exchanged at the correct times (same slab boundaries). At that point, the worldtube singleton will definitely need the correct global time step.

Reply:
OK.  That makes sense.  Thanks for explaining.